app/models/throttle.rb

```diff
@@ -48,10 +49,50 @@ class Throttle < ApplicationRecord
     },
   }.freeze
 
+  # Either target or user must be supplied
+  # @param [Symbol] throttle_type
+  # @param [String] target
+  # @param [User] user
+  # @return [Throttle]
+  def self.for(throttle_type:, user: nil, target: nil)
+    throttle = if user
+      find_or_create_by(user: user, throttle_type: throttle_type)
+    elsif target
+      find_or_create_by(target: target, throttle_type: throttle_type)
+    else
+      raise 'Throttle must have a user or a target, but neither were provided'
+    end
+
+    throttle.tap do |t|
+      t.reset_if_expired_and_maxed
+    end
+  end
+
+  # @return [Throttle]
+  def increment
+    return self if maxed?
+    update(attempts: attempts + 1, attempted_at: Time.zone.now)
+    self
```

Comment:
Kinda curious about these returning the instance itself. It seems to have had existed previously, though unclear if we want to change that as part of the refactoring. We already are in the case of `throttled_else_increment?` which previously returned the throttle instance if it was throttled (now returns `true`).

Reply:
Yeah I kept this to match the old behavior, but I have no idea what an expected return value for `increment` would be... in other languages I'd make this `void` or return the count... I guess I'll have it return the count

Reply:
8db4821535307a30a6963b6325ead9cf7fa7f091